### PR TITLE
Update environments.rst

### PR DESCRIPTION
**Pull request by aphecetche**
I believe the current syntax in the doc is incorrect. At least it's not working for me...

### CODE REVIEW

File: lib/spack/docs/environments.rst [hunk 1]
```diff
@@ -716,7 +716,7 @@ named list ``compilers`` is ``['%gcc', '%clang', '%intel']`` on
    spack:
      definitions:
        - compilers: ['%gcc', '%clang']
-       - when: arch.satisfies('x86_64:')
+         when: arch.satisfies('x86_64:')
```
Comment (alalazo) [inline lib/spack/docs/environments.rst:719]: Hmm, this seems to be a wrong fix. What was failing in your case, when following the example?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: lib/spack/docs/environments.rst (hunk 1 above)
Comment (aphecetche) [reply lib/spack/docs/environments.rst:719]: to be fair, the error was not strictly in following the example, but trying to get the same package complete spec be different on different platforms : 

```
📦spack-tutorial.../spack-tutorial/errors [spack-tutorial] $ cat -p spack.yaml
spack:
  definitions:
  - root_spec: [root +x +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
  - when: arch.satisfies('platform=linux')
  - root_spec: [root +aqua ~x ~tbb +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
  - when: arch.satisfies('platform=darwin')
  specs:
  - $root_spec
  concretizer:
    unify: true
📦spack-tutorial.../spack-tutorial/errors [spack-tutorial] $ spack env activate .
==> Error:
```

```
📦ok...iles/spack-tutorial/ok [spack-tutorial] $ cat -p spack.yaml
spack:
  definitions:
  - root_spec: [root +x +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
    when: arch.satisfies('platform=linux')
  - root_spec: [root +aqua ~x ~tbb +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
    when: arch.satisfies('platform=darwin')
  specs:
  - $root_spec
  concretizer:
    unify: true
📦ok...iles/spack-tutorial/ok [spack-tutorial] $ spack env activate .
(no error here)
```

Comment (alalazo) [reply lib/spack/docs/environments.rst:719]: I agree, but that is exactly my point. The first time you wrote it wrong, i.e. you have a YAML list with 4 items (and some items have just the `when` attribute, which makes them wrong):
```yaml
definitions:
  - root_spec: [root +x +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
  - when: arch.satisfies('platform=linux')
  - root_spec: [root +aqua ~x ~tbb +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
  - when: arch.satisfies('platform=darwin')
```
The second time you wrote it like in the docs and it works (only 2 items):
```yaml
definitions:
  - root_spec: [root +x +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
    when: arch.satisfies('platform=linux')
  - root_spec: [root +aqua ~x ~tbb +opengl +http +pythia6 +spectrum generator=ninja cxxstd=17]
    when: arch.satisfies('platform=darwin')
```
So, in this PR you're introducing a typo :slightly_smiling_face:

Comment (aphecetche) [reply lib/spack/docs/environments.rst:719]: hum. I'm confused. this PR is removing a `-`, not adding it, isn't it ? I must be missing something. The `when` is applying to the item above it, or below it ? 
In the doc https://spack.readthedocs.io/en/latest/environments.html  

Capture d’écran 2024-01-30 à 10 48 47

Comment (alalazo) [reply lib/spack/docs/environments.rst:719]: The `when` in the doc is in the same YAML object as the `compilers` below it. Note that here we are discussing YAML parsing, not anything that is Spack specific.

Comment (aphecetche) [reply lib/spack/docs/environments.rst:719]: ok, so it was just me confused by YAML. Thanks for the explanations. Closing this PR then.

**Pull request closed without merging**